samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinatorFactory.java

```diff
@@ -33,6 +33,7 @@
 public class ZkJobCoordinatorFactory implements JobCoordinatorFactory {
 
   private static final Logger LOG = LoggerFactory.getLogger(ZkJobCoordinatorFactory.class);
+  private static final String JOB_COORDINATOR_ZK_PATH_SUFFIX = "/JobCoordinatorData";
```

Comment:
Yes. We should refactor them. Within `ZkJobCoordinator`, we tried using a specific `ZkKeyBuilder` class. Not sure how to re-use the same key building utilities if they are in logically different components. But we can figure it out when we refactor our APIs.